tests/callbacks/test_early_stopper.py

```diff
@@ -0,0 +1,72 @@
+from typing import List, Sequence
+
+import pytest
+from torch import tensor
+from torch.utils.data import DataLoader
+
+from composer import Trainer
+from composer.callbacks.early_stopper import EarlyStopper
+from composer.core import State
+from composer.core.callback import Callback
+from composer.core.time import Time, TimeUnit
+from composer.loggers import Logger
+from tests.common import RandomClassificationDataset, SimpleModel
+
+
+class TestMetricSetter(Callback):
+
+    def __init__(self, monitor: str, dataloader_label: str, metric_sequence: Sequence, unit: TimeUnit):
+        self.monitor = monitor
+        self.dataloader_label = dataloader_label
+        self.metric_sequence = metric_sequence
+        self.unit = unit
+
+    def _update_metrics(self, state: State):
+        idx = min(len(self.metric_sequence) - 1, state.timer.get(self.unit).value)
+        metric_val = self.metric_sequence[idx]
+        state.current_metrics[self.dataloader_label] = state.current_metrics.get(self.dataloader_label, dict())
+        state.current_metrics[self.dataloader_label][self.monitor] = tensor(metric_val)
```

Comment:
(See below comment on gpu tests), but can you create this tensor on the same device as `next(torch.model.parameters()).device`? I believe that a gpu tensor will be stored in state, if using gpu. Would like to test this codepath.